src/main/webapp/app/exercises/programming/shared/utils/programming-exercise.utils.ts

```diff
@@ -79,8 +79,8 @@ export const isResultPreliminary = (latestResult: Result, programmingExercise?:
     if (resultCompletionDate && !dayjs.isDayjs(resultCompletionDate)) {
         resultCompletionDate = dayjs(resultCompletionDate);
     }
-    // When the result completionDate would be null, we have to return here (edge case, every result should have a completionDate).
-    if (!resultCompletionDate || !resultCompletionDate.isValid()) {
+    // When the result completionDate is invalid, we have to return here (edge case, every result should have a valid completionDate).
+    if (!resultCompletionDate.isValid()) {
```

Comment:
This seems dangerous. I cannot think about a case where the resultDate might not be set, but I don't see the benefit of risking a crash if it does happen.

Reply:
Generally I would agree with you. I just removed it because `!resultCompletionDate` will always evaluate to `false` because the previous if condition returns from the function if it would evaluate to `true`:
![image](https://user-images.githubusercontent.com/12861668/139587574-9e39a5a3-3110-45a9-abed-cbeaac91c160.png)

But yeah, I also get your argument. Never change a running system. So I am also fine with leaving the code as it was before.